app/views/notify-token-drawer/notify-email-form.tsx

```diff
@@ -0,0 +1,76 @@
+import type { ApolloError } from '@apollo/client'
+import type { FormEventHandler } from 'react'
+import { useForm } from 'react-hook-form'
+import clsx from 'clsx'
+import styles from './wallet-linking-drawer.module.scss'
+import { FormInput } from '@/components/form-input/form-input'
+import Button from '@/components/button/button'
+import { emailPattern } from '@/validators'
+import IconEmail from '@/components/icons/icon-email'
+
+export interface NotifyEmailFormInput {
+  email: string
+}
+
+export interface NotifyEmailFormProps {
+  onSubmit: (data: NotifyEmailFormInput) => void | Promise<void>
+  success?: boolean
+  loading?: boolean
+  error?: Error | ApolloError | null
+}
+
+export function NotifyEmailForm({
+  onSubmit,
+  loading,
+  success,
+  error,
+}: NotifyEmailFormProps) {
+  const { control, handleSubmit, formState } = useForm<NotifyEmailFormInput>()
+
+  const fullFormDisabled = loading || success
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit) as FormEventHandler<any>} className={styles.form}>
+      <div className={styles.header}>
+        <h3>Thank you!<br/>We can notify you when we add it</h3>
+      </div>
+      <FormInput
```

Comment:
Should we move email input to separate file and reuse it in 3 forms?

Reply:
Yep, it will be better